src/libraries/Microsoft.PowerFx.Core/Types/Enums/BuiltInEnums.cs

```diff
@@ -48,7 +49,8 @@ internal static class BuiltInEnums
                 { "LongDateTime24", "'longdatetime24'" },
                 { "ShortDateTime24", "'shortdatetime24'" },
                 { "UTC", "utc" }
-            });
+            },
+            canCoerceFromBackingKind: true);
```

Comment:
>canCoerceFromBackingKind: true [](http://example.com/codeflow?start=12&length=30)

If I understand correctly, that will allow us to create this enum from the string such as "utc" or "'shortdate'". Is that the case?

Reply:
Kind of.  From a type system perspective, this accurately describing the semantics for the second argument to Text (that it should be this enum and not another) and that we do want strings to be convertible to DateTimeFormat.CalculatedOptionSetValue for custom formats.  An implementation can check to see if they get back that enum value (or some other bit that says it was calculated) and treat it like a string instead of the enum value.

But in the current implementation, it already checks for enum values separaete from strings, before that coercion would happen.  It probably could be rewritten to use this mechanism and be less special case.

Thinking about IntelliSense, this might be important, but taking that even further we should probably have a bit for "MustBeConstant" too, both for this and for the Match enum, so that we don't recommend variables that are string valued.  I'm not sure how IntelliSense works and if it uses CheckTypes, perhaps this is already covered.

In any case, I have added some tests.  For Pre-V1, the interpreter doesn't do the right thing, hence the override, but the interpreter shouldn't be used in Pre-V1 mode anyway.
https://github.com/microsoft/Power-Fx/blob/5aff062b5fc04ceebab980578fb315f080bd3d31/src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/StronglyTypedEnum_BuiltInEnums.txt#L1192